modules/microshift-data-cleaning-container-images.adoc

```diff
@@ -0,0 +1,95 @@
+
+Keeping the container images helps speed up {microshift-short} restart after data clean up because the necessary container images are already present locally when you start the service.
+
+When you run the script with the `--all` and `--keep-images` arguments, you perform the following clean up actions:
+
+* Stop and disable all {microshift-short} services
+* Delete all {microshift-short} pods
+* Reset network configuration
+* Delete the `/var/lib/microshift` data directory
+* Delete OVN-K networking configuration
+
+[WARNING]
+====
+This option deletes all {microshift-short} data and user workloads. Use with caution.
+====
+
+.Prerequisites
+* You are logged into {microshift-short} as an administrator with root-user access.
+* You have filed a support case.
+
+.Procedure
+
+. Clean up all data and user workloads while retaining the {microshift-short} container images by running the `microshift-cleanup-data` script with the `--all` and `--keep-images` argument, by entering the following command:
++
+[source,terminal]
+----
+$ sudo microshift-cleanup-data --all --keep-images
+----
++
+.Example output
+[source,terminal]
+----
+DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?
+1) Yes
+2) No
+#? Yes
+Stopping MicroShift services
+Disabling MicroShift services
+Removing MicroShift pods
+Deleting the br-int interface
+Killing conmon, pause and OVN processes
+Removing MicroShift configuration
+Removing OVN configuration
+MicroShift service was stopped
+MicroShift service was disabled
+Cleanup succeeded
+----
+. Verify that the container images are still present by running the following command:
++
+[source,terminal]
+----
+$ sudo crictl images | awk '{print $1}'
+----
++
+.Example output
+[source,terminal]
+----
+IMAGE
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+quay.io/openshift-release-dev/ocp-v4.0-art-dev
+registry.redhat.io/lvms4/topolvm-rhel9
+registry.redhat.io/openshift4/ose-csi-external-provisioner
+registry.redhat.io/openshift4/ose-csi-external-resizer
+registry.redhat.io/openshift4/ose-csi-livenessprobe
+registry.redhat.io/openshift4/ose-csi-node-driver-registrar
+registry.redhat.io/ubi9
+----
++
+[IMPORTANT]
+====
+The {microshift-short} service is stopped and disabled after you run the script.
+====
+. Restart the {microshift-short} service by running the following command:
++
+[source,terminal]
+----
+$ sudo systemctl enable --now microshift
```

Comment:
@ggiguash Do we want to point the user to the location of the newly generated kubeconfig files after the procedure?   or is that well-covered elsewhere?

Reply:
It is documented [here](https://docs.redhat.com/en/documentation/red_hat_build_of_microshift/4.16/html/installing/microshift-install-rpm#accessing-microshift-cluster_microshift-install-rpm)

If it is not "obvious" that deleting the whole database would necessitate update of kubeconfig to access the new cluster, we may want to add it, yes.